ribasim_qgis/core/nodes.py

```diff
@@ -123,6 +123,15 @@ def set_dropdown(self, name: str, options: set[str]) -> None:
         )
         layer.setEditorWidgetSetup(index, setup)
 
+    def set_unique(self, name: str) -> None:
+        layer = self.layer
+        index = layer.fields().indexFromName(name)
+        setup = QgsEditorWidgetSetup(
```

Comment:
I tried to keep a separation between Qgis things (in the widgets folder), and core things that can be unit tested. It feels like this is being compromised with this new code as codecode also tells us there are no unit tests.